File: boj/2873.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int arr[][] = new int[1003][1003];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int r = Integer.parseInt(st.nextToken());
        int c = Integer.parseInt(st.nextToken());
        int minVal = 9999, minRow = 0, minCol = 0;
        for (int i = 0; i < r; ++i) {
            st = new StringTokenizer(br.readLine());
            for (int j = 0; j < c; ++j) {
                arr[i][j] = Integer.parseInt(st.nextToken());
                if ((i + j) % 2 == 1 && minVal > arr[i][j]) {
                    minVal = arr[i][j];
                    minRow = i;
                    minCol = j;
                }
            }
        }
        StringBuilder sb = new StringBuilder();
        if (r % 2 != 0) {
            for (int i = 0; i < r; ++i) {
                for (int j = 0; j < c - 1; ++j) {
                    if (i % 2 == 0) sb.append("R");
                    else sb.append("L");
                }
                if(i != r - 1) sb.append("D");
            }
        } else if (c % 2 != 0) {
            for (int i = 0; i < c; ++i) {
                for(int j = 0; j < r - 1; ++j) {
                    if (i % 2 == 0) sb.append("D");
                    else sb.append("U");
                }
                if (i != c - 1) sb.append("R");
            }
        } else {
            int row = (minRow % 2 == 1) ? minRow - 1 : minRow;
            for (int i = 0; i < row; ++i) {
                for (int j = 0; j < c - 1; ++j) {
                    if (i % 2 == 0) sb.append("R");
                    else sb.append("L");
                }
                if (i != r - 1) sb.append("D");
            }
            boolean flag = true;
            for (int i = 0; i < c; ++i) {
                for (int j = row; j < row + 1; ++j) {
                    if (i == minCol) {
                        continue;
                    } else {
                        if (flag) sb.append("D");
                        else sb.append("U");
                        flag = !flag;
                    }
                    
                }
                if (i != c - 1) sb.append("R");
            }
            row += 2;
            if(row < r) sb.append("D");
            for (int i = row; i < r; ++i) {
                for (int j = 0; j < c - 1; ++j) {
                    if (i % 2 == 0) sb.append("L");
                    else sb.append("R");
                }
                if (i != r - 1) sb.append("D");
            }
        }
        sb.append("\n");
        System.out.println(sb);
    }
}

File: boj/2873.cpp
#include <iostream>

using namespace std;

int arr[1003][1003];

int main(void) {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);
	int r, c; cin >> r >> c;
	int min_val = 9999, min_r, min_c;
	for (int i = 0; i < r; ++i) {
		for (int j = 0; j < c; ++j) {
			cin >> arr[i][j];
			if ((i + j) % 2 == 1 && min_val > arr[i][j]) {
				min_val = arr[i][j];
				min_r = i;
				min_c = j;
			}
		}
	}
	if (r % 2 != 0) {
		for (int i = 0; i < r; ++i) {
			for (int j = 0; j < c - 1; ++j) {
				if (i % 2 == 0) cout << "R";
				else cout << "L";
			}
			if(i != r - 1) cout << "D";
		}
	} else if (c % 2 != 0) {
		for (int i = 0; i < c; ++i) {
			for(int j = 0; j < r - 1; ++j) {
				if (i % 2 == 0) cout << "D";
				else cout << "U";
			}
			if (i != c - 1) cout << "R";
		}
	} else {
		int row = min_r % 2 ? min_r - 1 : min_r;
		for (int i = 0; i < row; ++i) {
			for (int j = 0; j < c - 1; ++j) {
				if (i % 2 == 0) cout << "R";
				else cout << "L";
			}
			if (i != r - 1) cout << "D";
		}
		bool flag = true;
		for (int i = 0; i < c; ++i) {
			for (int j = row; j < row + 1; ++j) {
				if (i == min_c) {
					continue;
				} else {
					if (flag) cout << "D";
					else cout << "U";
					flag = !flag;
				}
				
			}
			if (i != c - 1) cout << "R";
		}
		row += 2;
		if(row < r) cout << "D";
		for (int i = row; i < r; ++i) {
			for (int j = 0; j < c - 1; ++j) {
				if (i % 2 == 0) cout << "L";
				else cout << "R";
			}
			if (i != r - 1) cout << "D";
		}
	}
	return 0;
}